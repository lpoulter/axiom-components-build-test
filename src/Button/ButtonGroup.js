import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';
import InlineGroup from '../InlineGroup/InlineGroup';

export default class ButtonGroup extends Component {
  static propTypes = {
    /** Content inserted into the group, ideally Buttons */
    children: PropTypes.node.isRequired,
    /** Whether the child Buttons should be joined */
    joined: PropTypes.bool,
  };

  static childContextTypes = {
    joined: PropTypes.bool,
  };

  getChildContext() {
    return {
      joined: this.props.joined,
    };
  }

  render() {
    const { children, joined, ...rest } = this.props;

    return (
      <Base space="x6" { ...rest }
          Component={ joined ? 'div' : InlineGroup }
          className="ax-button-group"
          textBreak={ joined ? 'none' : null }>
        { children }
      </Base>
    );
  }
}



