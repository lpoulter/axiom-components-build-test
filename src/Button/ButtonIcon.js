import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Icon from '../Icon/Icon';
import './ButtonIcon.css';

export const ButtonIconRef = 'ButtonIcon';

export default class ButtonIcon extends Component {
  static propTypes = {
    /** Color of the Icon */
    color: PropTypes.string,
    /** SKIP */
    isEnd: PropTypes.bool,
    /** SKIP */
    isStart: PropTypes.bool,
    /** Name of the Icon */
    name: PropTypes.string.isRequired,
    /** Size of the Icon */
    size: PropTypes.string,
  };

  static defaultProps = {
    size: '1rem',
  };

  static typeRef = ButtonIconRef;

  render() {
    const { isEnd, isStart, name, size, color, ...rest } = this.props;
    const classes = classnames('ax-button__icon', {
      'ax-button__icon--start': isStart,
      'ax-button__icon--end': isEnd,
    });

    return (
      <span { ...rest } className={ classes }>
        <Icon name={ name } size={ size } textColor={ color }/>
      </span>
    );
  }
}
