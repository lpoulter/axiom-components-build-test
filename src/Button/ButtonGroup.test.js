import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';
import ButtonGroup from './ButtonGroup';

const getComponent = (props = {}) =>
  renderer.create(
    <ButtonGroup { ...props }>
      <Button>Lorem ipsum</Button>
    </ButtonGroup>
  );

describe('ButtonGroup', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
