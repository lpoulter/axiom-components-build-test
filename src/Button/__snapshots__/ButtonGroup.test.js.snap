// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders with defaultProps 1`] = `
<div
  className="ax-inline-group ax-button-group ax-space--x6"
>
  <div
    className="ax-inline-group__spacer"
  >
    <button
      className="ax-button ax-button--accent ax-button--primary ax-button--rectangle-medium"
    >
      Lorem ipsum
    </button>
  </div>
</div>
`;
